### DISCUSSION TITLE
I Can't Disable Vercel Data Cache?

### DISCUSSION THREAD

**Discussion by OatsProgramming** [vercel/next.js#51455] [Help]
### Summary

*Note: this is a repost of this [discussion](https://github.com/vercel/vercel/discussions/10117#discussion-5309327) since there are more people here in vercel/next.js than there is in vercel/vercel*

Im trying to get around its “automatic caching” system since its giving me stale data every time. 

I’ve set cache: no-store and no-cache and that didnt work
next: { revalidate: 0 } and that still gave me old data:

*I've added this at the top lvl of the module & in my specific route:*
```ts
export const fetchCache = 'force-no-store';
```

*I've done the following for my specific fetch:*
```ts
const res = await fetch('api/example', { cache: 'no-store' });
const res = await fetch('api/example', { cache: 'no-cache' });
const res = await fetch('api/example', { next: { revalidate: 0 });
const res = await fetch('api/example', { cache: 'no-store', next: { revalidate: 0 });
const res = await fetch('api/example', { cache: 'no-cache', next: { revalidate: 0 });
```

Checked their docs for Vercel Caching and did the following for my backend: 

```ts
let cacheControl = 'public, s-maxage=1000, stale-while-revalidate=500'
        if (toCache.has(slug[0])) cacheControl = 'public, s-maxage=3600, stale-while-revalidate=1800'

        return new Response(stringifyResult, {
            status: 200, 
            headers: {
                'Content-Type': 'application/json',
                'Cache-Control': cacheControl,
                'Vercel-CDN-Cache-Control': 's-maxage=1000',
                'CDN-Cache-Control': 's-maxage=1000'
            }
        })
```

*Even though I've set it to simply a 1000s, I'd get stale data still by the next day. Set to 0 to see if it'll do anything: still the same result*

I've also added the following vercel.json file:
```json
{
    "headers": [
      {
        "source": "/api/INSERT_ROUTE/route.ts",
        "headers": [
          {
            "key": "Cache-Control",
            "value": "public, s-maxage=3600, stale-while-revalidate=1800"
          }
        ]
      },
      {
        "source": "/api/INSERT_ROUTE2/route.ts",
        "headers": [
          {
            "key": "Cache-Control",
            "value": "public, s-maxage=3600, stale-while-revalidate=1800"
          }
        ]
      }
    ]
  }
  ```

Im still getting stale data

Ive purged the vercel data cache (which worked temporarily as expected) but i’d get stale data the next day or whenever new data is supposed to come

**Comment by OatsProgramming** [comment]
After starting from scratch and making sure that there are no other variables, I've re-implemented the "solutions" to my problem.
```ts
// i.e.
const res = await fetch('api/example', { cache: 'no-store' });
const res = await fetch('api/example', { cache: 'no-cache' });
const res = await fetch('api/example', { next: { revalidate: 0 });
const res = await fetch('api/example', { cache: 'no-store', next: { revalidate: 0 });
const res = await fetch('api/example', { cache: 'no-cache', next: { revalidate: 0 });
```
I found that `fetch("api/example", { next: { revalidate: 0 }}` to be the way to disable Vercel's Data Cache. However, in the [docs](https://vercel.com/docs/infrastructure/data-cache/manage-data-cache), its stated: 

>  ## Disabling Vercel Data Cache
> Use the following in your Next.js application to opt out of fetch caching:
> `export const fetchCache = 'force-no-store';`
> You can also opt out individual fetches by using the cache option:
> `const res = await fetch('https://example.com', { cache: 'no-store' });`

However, this doesn't opt out of fetch caching. I'm not sure if I'm just doing something wrong or the documentation is, but all I know that is the solution provided didn't work. Does anyone have this problem?

**Reply by leesontungggg** [reply]
I'm having the same issue, tried all the ways in the docs but not working too.

**Comment by leerob** [comment]
Closing as a duplicate of https://github.com/vercel/vercel/discussions/10117.
